guides/plugins/plugins/administration/add-custom-service.md

```diff
@@ -99,4 +143,23 @@ Shopware.Application.addServiceProviderDecorator('joker', joker => {
 
     return joker;
 });
-```
\ No newline at end of file
+```
+```
+
+## Reset Providers
+
+Another useful BottleJS is the [`resetProviders`](https://github.com/young-steveo/bottlejs#resetprovidersnames) function.
+This function is used to reset providers for the next reference to re-instantiate the provider. 
+You would do this to add decorators or middleware to your services after you have instantiated them.
+
+```javascript
+Shopware.Application.$container.resetProviders()
+```
```

Comment:
I would also add a practical example where you add middleware afterwards to existing services from the core